packages/@lwc/ssr-compiler/src/compile-template/shared.ts

```diff
@@ -172,3 +177,11 @@ export function getChildAttrsOrProps(
 
     return b.objectExpression(objectAttrsOrProps);
 }
+
+/**
+ * Determine if the provided node is of type Literal
+ * @param node
+ */
+export function isLiteral(node: IrLiteral | IrExpression | IrComplexExpression): node is IrLiteral {
```

Comment:
```suggestion
export function isLiteral(node: IrNode): node is IrLiteral {
```

Reply:
No need to enumerate all possible inputs; the function tells us the type.

Reply:
It seems `IrNode` doesn't encompass `IrLiteral | IrExpression | IrComplexExpression`.  I think it should be `IrBaseNode`, but `IrLiteral` and doesn't currently extend from it, so the change would need to be `IrBaseNode | IrLiteral`. But then I'm not sure if it works for `IrComplexExpression`:

```typescript
type ComplexExpression = /* AcornNode */ {
    start: number;
    end: number;
    type: string;
    range?: [number, number] | undefined;
    loc?: SourceLocation | null;
} & {
    value?: any;
}
```

All this to say, it might be better to leave it as is and refactor the typings in template-compiler if possible 🤷

Reply:
Seems like there probably would be some typing issues (Literal isn't BaseNode, etc). Perhaps we can review in a follow up?

Reply:
Sure, no prob.

Reply:
Created #5028